File: lib-database-entities/src/main/java/com/ustadmobile/lib/db/entities/DashboardEntryTag.java
package com.ustadmobile.lib.db.entities;

import com.ustadmobile.lib.database.annotation.UmEntity;
import com.ustadmobile.lib.database.annotation.UmPrimaryKey;
import com.ustadmobile.lib.database.annotation.UmSyncLastChangedBy;
import com.ustadmobile.lib.database.annotation.UmSyncLocalChangeSeqNum;
import com.ustadmobile.lib.database.annotation.UmSyncMasterChangeSeqNum;

@UmEntity(tableId = 82)
public class DashboardEntryTag {

    @UmPrimaryKey(autoGenerateSyncable = true)
    private long dashboardEntryTagUid;

    private long dashboardEntryTagDashboardEntryUid;

    private long dashboardEntryTagDashboardTagUid;

    private boolean dashboardEntryTagActive;

    @UmSyncMasterChangeSeqNum
    private long dashboardEntryTagMCSN;

    @UmSyncLocalChangeSeqNum
    private long dashboardEntryTagLCSN;

    @UmSyncLastChangedBy
    private int dashboardEntryTagLCB;

    public DashboardEntryTag(){
        dashboardEntryTagActive = true;
    }

    public DashboardEntryTag(long entryUid, long tagUid){
        dashboardEntryTagActive = true;
        this.dashboardEntryTagDashboardEntryUid = entryUid;
        this.dashboardEntryTagDashboardTagUid = tagUid;
    }

    public long getDashboardEntryTagUid() {
        return dashboardEntryTagUid;
    }

    public void setDashboardEntryTagUid(long dashboardEntryTagUid) {
        this.dashboardEntryTagUid = dashboardEntryTagUid;
    }

    public long getDashboardEntryTagDashboardEntryUid() {
        return dashboardEntryTagDashboardEntryUid;
    }

    public void setDashboardEntryTagDashboardEntryUid(long dashboardEntryTagDashboardEntryUid) {
        this.dashboardEntryTagDashboardEntryUid = dashboardEntryTagDashboardEntryUid;
    }

    public long getDashboardEntryTagDashboardTagUid() {
        return dashboardEntryTagDashboardTagUid;
    }

    public void setDashboardEntryTagDashboardTagUid(long dashboardEntryTagDashboardTagUid) {
        this.dashboardEntryTagDashboardTagUid = dashboardEntryTagDashboardTagUid;
    }

    public boolean isDashboardEntryTagActive() {
        return dashboardEntryTagActive;
    }

    public void setDashboardEntryTagActive(boolean dashboardEntryTagActive) {
        this.dashboardEntryTagActive = dashboardEntryTagActive;
    }

    public long getDashboardEntryTagMCSN() {
        return dashboardEntryTagMCSN;
    }

    public void setDashboardEntryTagMCSN(long dashboardEntryTagMCSN) {
        this.dashboardEntryTagMCSN = dashboardEntryTagMCSN;
    }

    public long getDashboardEntryTagLCSN() {
        return dashboardEntryTagLCSN;
    }

    public void setDashboardEntryTagLCSN(long dashboardEntryTagLCSN) {
        this.dashboardEntryTagLCSN = dashboardEntryTagLCSN;
    }

    public int getDashboardEntryTagLCB() {
        return dashboardEntryTagLCB;
    }

    public void setDashboardEntryTagLCB(int dashboardEntryTagLCB) {
        this.dashboardEntryTagLCB = dashboardEntryTagLCB;
    }
}

File: lib-database-entities/src/commonMain/kotlin/com/ustadmobile/lib/db/entities/DashboardEntryTag.kt
package com.ustadmobile.lib.db.entities

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.ustadmobile.door.annotation.LastChangedBy
import com.ustadmobile.door.annotation.LocalChangeSeqNum
import com.ustadmobile.door.annotation.MasterChangeSeqNum
import com.ustadmobile.door.annotation.SyncableEntity
import kotlinx.serialization.Serializable

@SyncableEntity(tableId = 83)
@Entity
@Serializable
class DashboardEntryTag {

    @PrimaryKey(autoGenerate = true)
    var dashboardEntryTagUid: Long = 0

    var dashboardEntryTagDashboardEntryUid: Long = 0

    var dashboardEntryTagDashboardTagUid: Long = 0

    var dashboardEntryTagActive: Boolean = false

    @MasterChangeSeqNum
    var dashboardEntryTagMCSN: Long = 0

    @LocalChangeSeqNum
    var dashboardEntryTagLCSN: Long = 0

    @LastChangedBy
    var dashboardEntryTagLCB: Int = 0

    constructor() {
        dashboardEntryTagActive = true
    }

    constructor(entryUid: Long, tagUid: Long) {
        dashboardEntryTagActive = true
        this.dashboardEntryTagDashboardEntryUid = entryUid
        this.dashboardEntryTagDashboardTagUid = tagUid
    }
}
